lib/xcvm/protobuf/src/xcvm.proto

```diff
@@ -2,6 +2,12 @@ syntax = "proto3";
 
 package xcvm;
 
+/// proto buf tells that
+/// - any repeated field can be intemingled with not not, e.g. `asset A, account, asset B`can happen
+/// - also can happend repeats, which are really not, exampeles `transfer, transfer` and `asset A, asset A`
+/// so proto are not enforce inter field order niether exact once by specfication,
+/// so that should be specified on XCVM compatible protocol to use proper protoc
```

Comment:
If the decoder implements the spec properly, it should be able to handle this case right? 

Basing my knowledge off [this](https://stackoverflow.com/questions/55063424/does-repeated-fields-in-protobuffers-keep-the-order-they-are-inserted).

Reply:
order of one field yes, but inter field is not guaranteed like in comment in above link https://stackoverflow.com/questions/55063424/does-repeated-fields-in-protobuffers-keep-the-order-they-are-inserted#comment96919403_55069102 

neither is guaranteed that field only encoded once even if singular.

so by spec proto is kind of of no ideal for linear interpretation with lowers possible memory usage. so if compilers used are more strict than spec, then that will work.